Source/Adapters/Blackjack.Adapters/UserAdapter.cs

```diff
@@ -54,6 +62,19 @@ public async Task<UserBE> GetUsers(string username)
         return usersBySoundex?.SingleOrDefault(user => user.Username.Equals(username, StringComparison.OrdinalIgnoreCase));
     }
 
+    public async Task<UserBE> UpdateUser(UserBE user)
+    {
+        UserBE dbUser = await GetUsers(user.Username);
+        if (dbUser == null)
+        {
+            throw new WebException(HttpStatusCode.BadRequest, "User does not exist.");
```

Comment:
You XML Doc comment above in your interface says you return null if the user doesn't exist which isn't true.

Update that comment.

This is the correct approach, if you trip a business validation everything should shut down